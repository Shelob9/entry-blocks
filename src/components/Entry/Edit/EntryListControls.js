import React, {createElement, Fragment} from 'react';
import {findFormById} from "../findFormById";

import {ChooseForm} from "../../controls/ChooseForm";
import {ChooseEntry} from "../../controls/ChooseEntry";

export default function (props) {
	const {
		entries,
		formId,
		entryId,
		onSetForm,
		onSetEntry,
		instanceId,
		forms
	} = props;
	let form = {};
	if (!props.form) {
		form = findFormById(forms, formId);
	} else {
		form = props.form;
	}

	const elements = [
		<ChooseForm
			forms={forms}
			currentFormId={formId}
			onSetForm={onSetForm}
			instanceId={instanceId}
			key={1}
		/>
	];

	if (formId) {
		elements.push(<ChooseEntry
			entries={entries}
			currentEntry={entryId}
			onSetEntry={onSetEntry}
			instanceId={instanceId}
			key={2}
		/>);
	}



	return createElement(Fragment, {}, elements);
}
