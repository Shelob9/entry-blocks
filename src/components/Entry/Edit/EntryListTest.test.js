import renderer from "react-test-renderer";
import React from "react";

import Enzyme, {mount,shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

import EntryListControls from './EntryListControls'

const _forms = require( '../../../__MOCK_DATA__/forms' )
const forms = Object.values(_forms );
const formId = 'CF5bde22b646ed0';
const form = _forms[formId];
const entries = require( '../../../__MOCK_DATA__/entries' );
const entry = entries[0];
const entryFieldId = 'fld_8768091';
const entryId = entry.id;
const instanceId = '54fds';



describe( 'EntryListControls component', () => {
	const mockFuncGeneric = jest.fn();
	it( 'Matches snapshot with all props', () => {
		expect(
			renderer.create(
				<EntryListControls
					entryId={entryId}
					entries={entries}
					onSetEntry={mockFuncGeneric}
					formId={formId}
					forms={forms}
					form={form}
					onSetForm={mockFuncGeneric}
					instanceId={instanceId}
				/>
			)
		).toMatchSnapshot();
	});

	it( 'Matches snapshot with no entry selected ', () => {
		expect(
			renderer.create(
				<EntryListControls
					entryId={''}
					entries={entries}
					onSetEntry={mockFuncGeneric}
					formId={formId}
					forms={forms}
					form={form}
					onSetForm={mockFuncGeneric}
					instanceId={instanceId}
				/>
			)
		).toMatchSnapshot();
	});

	it( 'Matches snapshot with no form selected ', () => {
		expect(
			renderer.create(
				<EntryListControls
					entryId={''}
					entries={undefined}
					onSetEntry={mockFuncGeneric}
					formId={''}
					forms={forms}
					form={{}}
					onSetForm={mockFuncGeneric}
					instanceId={instanceId}
				/>
			)
		).toMatchSnapshot();
	});

	test( 'Calls onSetForm',() => {
		const mockFn = jest.fn();
		const event = {
			preventDefault() {},
			target: { value: 'matched' }
		};
		const component = mount(
			<EntryListControls
				entryId={''}
				entries={undefined}
				onSetEntry={mockFuncGeneric}
				formId={''}
				forms={forms}
				form={{}}
				onSetForm={mockFn}
				instanceId={instanceId}
			/>);
		component.find('.caldera-forms-form-chooser').simulate('change', event);
		expect(mockFn.mock.calls[0][0]).toBe('matched');

	});

	test( 'Calls onSetEntry',() => {
		const mockFn = jest.fn();
		const event = {
			preventDefault() {},
			target: { value: 'matched' }
		};
		const component = mount(
			<EntryListControls
				entryId={''}
				entries={entries}
				onSetEntry={mockFn}
				formId={formId}
				forms={forms}
				form={form}
				onSetForm={mockFuncGeneric}
				instanceId={instanceId}
			/>);
		component.find('.caldera-forms-entry-chooser').simulate('change', event);
		expect(mockFn.mock.calls[0][0]).toBe('matched');

	});

});