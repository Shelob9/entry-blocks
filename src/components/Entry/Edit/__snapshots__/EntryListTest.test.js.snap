exports[`EntryListControls component Matches snapshot with all props 1`] = `
Array [
  <div>
    <label
        className=""
        htmlFor="caldera-forms-form-chooser-54fds">
        Choose Form
    </label>
    <select
        className="caldera-forms-form-chooser"
        id="caldera-forms-form-chooser-54fds"
        onChange={[Function]}
        value="CF5bde22b646ed0">
        <option>
             -- 
        </option>
        <option
            value="CF5bde22a843265">
            Not Contact Form
        </option>
        <option
            value="CF5bde22b646ed0">
            Contact Form
        </option>
    </select>
</div>,
  <label
    htmlFor="caldera-forms-entry-chooser-54fds">
    Choose Entry
</label>,
  <select
    className="caldera-forms-entry-chooser"
    id="caldera-forms-entry-chooser-54fds"
    onChange={[Function]}
    value="1">
    <option
        value={null}>
        --
    </option>
    <option
        value="1">
        1
    </option>
    <option
        value="2">
        2
    </option>
</select>,
]
`;

exports[`EntryListControls component Matches snapshot with no entry selected  1`] = `
Array [
  <div>
    <label
        className=""
        htmlFor="caldera-forms-form-chooser-54fds">
        Choose Form
    </label>
    <select
        className="caldera-forms-form-chooser"
        id="caldera-forms-form-chooser-54fds"
        onChange={[Function]}
        value="CF5bde22b646ed0">
        <option>
             -- 
        </option>
        <option
            value="CF5bde22a843265">
            Not Contact Form
        </option>
        <option
            value="CF5bde22b646ed0">
            Contact Form
        </option>
    </select>
</div>,
  <label
    htmlFor="caldera-forms-entry-chooser-54fds">
    Choose Entry
</label>,
  <select
    className="caldera-forms-entry-chooser"
    id="caldera-forms-entry-chooser-54fds"
    onChange={[Function]}
    value="">
    <option
        value={null}>
        --
    </option>
    <option
        value="1">
        1
    </option>
    <option
        value="2">
        2
    </option>
</select>,
]
`;

exports[`EntryListControls component Matches snapshot with no form selected  1`] = `
<div>
  <label
    className=""
    htmlFor="caldera-forms-form-chooser-54fds">
    Choose Form
  </label>
  <select
    className="caldera-forms-form-chooser"
    id="caldera-forms-form-chooser-54fds"
    onChange={[Function]}
    value="">
    <option>
       -- 
    </option>
    <option
      value="CF5bde22a843265">
      Not Contact Form
    </option>
    <option
      value="CF5bde22b646ed0">
      Contact Form
    </option>
  </select>
</div>
`;
